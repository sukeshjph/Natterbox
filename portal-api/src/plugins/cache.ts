export default class Cache {
  memoryCache

  constructor(memoryCache) {
    this.memoryCache = memoryCache
  }

  async get(id) {
    const result = await this.memoryCache.get(id)
    return result
  }

  async set(id, token) {
    const result = await this.memoryCache.set(id, token)
    return result
  }
}