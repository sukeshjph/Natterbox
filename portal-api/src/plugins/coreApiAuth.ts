import axios from "axios"

export default class CoreApiAuth {
  gatekeeperBaseUrl

  lassoBaseUrl

  constructor() {
    this.gatekeeperBaseUrl =
      "https://gatekeeper.redmatter-qa01.pub/token/sapien/organisation"
    this.lassoBaseUrl = "http://lasso.qa01.redmatter.com/auth/token/coreapi"
  }

  async getLassoToken(orgId, userId, token, logger) {
    const gatekeeperToken: any = await this.getGatekeeperApiToken(
      orgId,
      userId,
      token,
      logger,
    )

    const result = await axios
      .get(this.lassoBaseUrl, {
        headers: {
          Authorization: `Bearer ${gatekeeperToken.data.jwt}`,
        },
      })
      .catch(() => {
        logger.error("Problem getting Lasso Token")
        // throw new Error("Problem getting Lasso Token. Might be routes or vpn")
      })

    return result.data.access_token
  }

  async getGatekeeperApiToken(orgId, userId, token, logger) {
    const result = await axios
      .get(
        `${this.gatekeeperBaseUrl}/${orgId}/user/${userId}?scope=portal:admin`,
        {
          headers: {
            Authorization: `Bearer ${token}`,
          },
        },
      )
      .catch(() => {
        logger.error("Problem getting Lasso Token")
        // throw new Error("Problem getting Gatekeeper Token, might be expired JWT")
      })

    return result
  }
}
