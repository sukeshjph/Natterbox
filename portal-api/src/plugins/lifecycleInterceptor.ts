export default {
  // Fires whenever a GraphQL request is received from a client.
  requestDidStart(requestContext) {
    // Do not log introspection as it creates clutter
    if (requestContext.request.operationName !== "IntrospectionQuery") {
      requestContext.context.logger.debug(`Request started`, {
        meta: `${requestContext.request.query}`,
      })
    }

    return {
      willSendResponse({ response }) {
        if (response.errors)
          requestContext.context.logger.error("Error", {
            meta: response.errors,
          })
      },
    }
  },
}
