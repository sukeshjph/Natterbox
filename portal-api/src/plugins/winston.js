import winston from "winston"

const date = new Date().toISOString()

const logFormat = winston.format.printf(info => {
  return `${date} [${info.level}] : ${info.message}\n  ${JSON.stringify(
    info.meta,
    null,
    4,
  ) || ""}`
})

export default winston.createLogger({
  level: process.env.LOG_LEVEL,
  format: (function format() {
    switch (process.env.LOG_FORMAT) {
      case "json":
        return winston.format.json()
      case "text":
        return winston.format.combine(
          winston.format.colorize(),
          winston.format.simple(),
          logFormat,
        )
      default:
        return winston.format.json()
    }
  })(),
  transports: [
    new winston.transports.Console(),
    new winston.transports.File({ filename: "log" }),
  ],
})
