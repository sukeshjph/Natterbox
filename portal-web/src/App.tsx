import React from "react"
import { ApolloProvider } from "@apollo/react-hooks"

import {
  createMuiTheme,
  ThemeProvider,
  StylesProvider,
} from "@material-ui/core/styles"
import { Route } from "react-router-dom"

// Our Libs
import "./styles/core-styles.scss"
import { Loading, Header, Footer, ErrorAlert } from "./components/shared"
import { AppRoutes } from "./AppRoutes"
import { useApp } from "./useApp.hook"

const theme = createMuiTheme({
  typography: {
    fontFamily: ["HelveticaNeue", "Roboto", "Arial"].join(","),
  },
})

const App = () => {
  const {
    isInitializing,
    isAuthenticated,
    loginWithRedirect,
    apolloLink,
    alerts,
    setAlerts,
    alertsOpen,
    setAlertsOpen,
  } = useApp()

  if (isInitializing) {
    return <Loading />
  }

  if (!isAuthenticated) {
    loginWithRedirect({})
  }

  return (
    <StylesProvider injectFirst>
      <ThemeProvider theme={theme}>
        <ApolloProvider client={apolloLink}>
          <>
            <div className="content">
              <Route
                path="/"
                render={({ location }) => (
                  <>
                    <Header location={location} />
                    {alerts.length > 0 &&
                      alertsOpen &&
                      alerts.map(
                        ({ message, severity, orgId, requestId, errors }) => (
                          <ErrorAlert
                            message={message}
                            severity={severity}
                            orgId={orgId}
                            requestId={requestId}
                            errors={errors}
                            onClose={() => setAlertsOpen(false)}
                          />
                        ),
                      )}
                    <AppRoutes setAlerts={setAlerts} />
                  </>
                )}
              />
            </div>
            <Footer />
          </>
        </ApolloProvider>
      </ThemeProvider>
    </StylesProvider>
  )
}

export default App
