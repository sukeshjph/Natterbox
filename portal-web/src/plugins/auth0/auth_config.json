{
  "domain": "redmatter-qa01.eu.auth0.com",
  "clientId": "sbi7zg4cJX6N8Io0tLtb6dBpom8YBXnq",
  "audience": "https://sapien-proxy.redmatter-qa01.pub/"
}
