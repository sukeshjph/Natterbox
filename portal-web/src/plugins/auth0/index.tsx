/* eslint-disable camelcase */
import React, { useState, useEffect, useContext } from "react"
import createAuth0Client from "@auth0/auth0-spa-js"
import Auth0Client from "@auth0/auth0-spa-js/dist/typings/Auth0Client"

export interface Auth0RedirectState {
  targetUrl?: string
}

export interface Auth0User extends Omit<IdToken, "__raw"> {}

interface Auth0Context {
  user?: Auth0User
  userToken?: string
  isAuthenticated: boolean
  isInitializing: boolean
  isPopupOpen: boolean
  apiUrl: string
  sapienUrl: string
  loginWithPopup(o?: PopupLoginOptions): Promise<void>
  handleRedirectCallback(): Promise<RedirectLoginResult>
  getIdTokenClaims(o?: getIdTokenClaimsOptions): Promise<IdToken>
  loginWithRedirect(o?: RedirectLoginOptions): Promise<void>
  getTokenSilently(o?: GetTokenSilentlyOptions): Promise<string | undefined>
  getTokenWithPopup(o?: GetTokenWithPopupOptions): Promise<string | undefined>
  logout(o?: LogoutOptions): void
}
interface Auth0ProviderOptions {
  children: React.ReactElement
  onRedirectCallback(result: RedirectLoginResult): void
}

export const Auth0Context = React.createContext<Auth0Context | null>(null)
export const useAuth0 = () => useContext(Auth0Context)!

const getAppendedSapienUrl = url => {
  if (!url.includes("v1")) {
    return url.slice(url.length - 1) === "/" ? `${url}v1/` : `${url}/v1/`
  }
  return url
}

const getConfigResponse = async () => {
  if (process.env.NODE_ENV === "development") {
    return {
      graphqlUrl: process.env.REACT_APP_API_URL,
      domain: process.env.REACT_APP_AUTH0_DOMAIN,
      client_id: process.env.REACT_APP_AUTH0_CLIENTID,
      audience: process.env.REACT_APP_AUTH0_AUDIENCE,
      sapien_url: process.env.REACT_APP_SAPIEN_URL,
    }
  }

  let response
  try {
    response = await fetch(`${window.location.origin}/config`)
  } catch (error) {
    throw new Error(
      `Failed to fetch config: Status:${response.status}, message:${error.message}`,
    )
  }

  const configData = await response.json()
  const { authDomain, clientId, sapienUrl, ...others } = configData

  return {
    ...others,
    domain: authDomain,
    client_id: clientId,
    sapien_url: getAppendedSapienUrl(sapienUrl),
  }
}

export const Auth0Provider = ({
  children,
  onRedirectCallback,
  ...initOptions
}: Auth0ProviderOptions & Pick<Auth0ClientOptions, "redirect_uri">) => {
  const [isAuthenticated, setIsAuthenticated] = useState(false)
  const [isInitializing, setIsInitializing] = useState(true)
  const [isPopupOpen, setIsPopupOpen] = useState(false)
  const [user, setUser] = useState<Auth0User>()
  const [userToken, setUserToken] = useState<"">()
  const [auth0Client, setAuth0Client] = useState<Auth0Client>()
  const [apiUrl, setApiUrl] = useState("")
  const [sapienUrl, setSapienUrl] = useState("")

  useEffect(() => {
    const initAuth0 = async () => {
      const {
        audience,
        domain,
        client_id,
        graphqlUrl,
        sapien_url,
      } = await getConfigResponse()

      const auth0FromHook = await createAuth0Client({
        ...initOptions,
        domain,
        client_id,
        audience,
      })

      setAuth0Client(auth0FromHook)
      setApiUrl(graphqlUrl)
      setSapienUrl(sapien_url)

      if (window.location.search.includes("code=")) {
        let appState: RedirectLoginResult = {}
        try {
          ;({ appState } = await auth0FromHook.handleRedirectCallback())
        } finally {
          onRedirectCallback(appState)
        }
      }

      const authed = await auth0FromHook.isAuthenticated()

      if (authed) {
        const userProfile = await auth0FromHook.getUser()
        const token = await auth0FromHook.getTokenSilently()

        setIsAuthenticated(true)
        setUser(userProfile)
        setUserToken(token)
      }

      setIsInitializing(false)
    }

    initAuth0()
  }, [])

  const loginWithPopup = async (options?: PopupLoginOptions) => {
    setIsPopupOpen(true)

    try {
      await auth0Client!.loginWithPopup(options)
    } catch (error) {
      console.error(error) //eslint-disable-line
    } finally {
      setIsPopupOpen(false)
    }

    const userProfile = await auth0Client!.getUser()
    setUser(userProfile)
    setIsAuthenticated(true)
  }

  const handleRedirectCallback = async () => {
    setIsInitializing(true)

    const result = await auth0Client!.handleRedirectCallback()
    const userProfile = await auth0Client!.getUser()

    setIsInitializing(false)
    setIsAuthenticated(true)
    setUser(userProfile)

    return result
  }

  const loginWithRedirect = (options?: RedirectLoginOptions) =>
    auth0Client!.loginWithRedirect(options)

  const getTokenSilently = (options?: GetTokenSilentlyOptions) =>
    auth0Client!.getTokenSilently(options)

  const logout = (options?: LogoutOptions) => auth0Client!.logout(options)

  const getIdTokenClaims = (options?: getIdTokenClaimsOptions) =>
    auth0Client!.getIdTokenClaims(options)

  const getTokenWithPopup = (options?: GetTokenWithPopupOptions) =>
    auth0Client!.getTokenWithPopup(options)

  return (
    <Auth0Context.Provider
      value={{
        user,
        isAuthenticated,
        isInitializing,
        isPopupOpen,
        userToken,
        loginWithPopup,
        loginWithRedirect,
        logout,
        getTokenSilently,
        handleRedirectCallback,
        getIdTokenClaims,
        getTokenWithPopup,
        apiUrl,
        sapienUrl,
      }}>
      {children}
    </Auth0Context.Provider>
  )
}
