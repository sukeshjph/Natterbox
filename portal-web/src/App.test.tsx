/* eslint-disable import/no-extraneous-dependencies */
import React from "react"
import { shallow } from "enzyme"
import toJson from "enzyme-to-json"
import App from "./App"

import { useAuth0 } from "./plugins/auth0"

jest.mock("./plugins/auth0")

jest.mock("react-router-dom", () => ({
  ...jest.requireActual("react-router-dom"),
  useHistory: () => ({
    push: jest.fn(),
  }),
}))

describe("Main App", () => {
  const setup = (mockOverrides = {}) => ({
    isInitializing: false,
    isAuthenticated: true,
    loginWithRedirect: () => undefined,
    userToken: "anyToken",
    ...mockOverrides,
  })

  it("Renders App", () => {
    useAuth0.mockReturnValueOnce(setup())
    const wrapper = shallow(<App />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })

  describe("App states", () => {
    it("initializing", () => {
      useAuth0.mockReturnValueOnce(setup({ isInitializing: true }))
      const wrapper = shallow(<App />)
      expect(wrapper.find(".spinPanel")).toBeDefined()
    })

    it("is not authenticated", () => {
      const redirectMock = jest.fn()
      useAuth0.mockReturnValueOnce(
        setup({ isAuthenticated: false, loginWithRedirect: redirectMock }),
      )
      shallow(<App />)
      expect(redirectMock).toHaveBeenCalled()
    })
  })
})
