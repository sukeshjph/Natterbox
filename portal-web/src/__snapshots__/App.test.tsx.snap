// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main App Renders App 1`] = `
<StylesProvider
  injectFirst={true}
>
  <ThemeProvider
    theme={
      Object {
        "breakpoints": Object {
          "between": [Function],
          "down": [Function],
          "keys": Array [
            "xs",
            "sm",
            "md",
            "lg",
            "xl",
          ],
          "only": [Function],
          "up": [Function],
          "values": Object {
            "lg": 1280,
            "md": 960,
            "sm": 600,
            "xl": 1920,
            "xs": 0,
          },
          "width": [Function],
        },
        "direction": "ltr",
        "mixins": Object {
          "gutters": [Function],
          "toolbar": Object {
            "@media (min-width:0px) and (orientation: landscape)": Object {
              "minHeight": 48,
            },
            "@media (min-width:600px)": Object {
              "minHeight": 64,
            },
            "minHeight": 56,
          },
        },
        "overrides": Object {},
        "palette": Object {
          "action": Object {
            "activatedOpacity": 0.12,
            "active": "rgba(0, 0, 0, 0.54)",
            "disabled": "rgba(0, 0, 0, 0.26)",
            "disabledBackground": "rgba(0, 0, 0, 0.12)",
            "disabledOpacity": 0.38,
            "focus": "rgba(0, 0, 0, 0.12)",
            "focusOpacity": 0.12,
            "hover": "rgba(0, 0, 0, 0.04)",
            "hoverOpacity": 0.04,
            "selected": "rgba(0, 0, 0, 0.08)",
            "selectedOpacity": 0.08,
          },
          "augmentColor": [Function],
          "background": Object {
            "default": "#fafafa",
            "paper": "#fff",
          },
          "common": Object {
            "black": "#000",
            "white": "#fff",
          },
          "contrastThreshold": 3,
          "divider": "rgba(0, 0, 0, 0.12)",
          "error": Object {
            "contrastText": "#fff",
            "dark": "#d32f2f",
            "light": "#e57373",
            "main": "#f44336",
          },
          "getContrastText": [Function],
          "grey": Object {
            "100": "#f5f5f5",
            "200": "#eeeeee",
            "300": "#e0e0e0",
            "400": "#bdbdbd",
            "50": "#fafafa",
            "500": "#9e9e9e",
            "600": "#757575",
            "700": "#616161",
            "800": "#424242",
            "900": "#212121",
            "A100": "#d5d5d5",
            "A200": "#aaaaaa",
            "A400": "#303030",
            "A700": "#616161",
          },
          "info": Object {
            "contrastText": "#fff",
            "dark": "#1976d2",
            "light": "#64b5f6",
            "main": "#2196f3",
          },
          "primary": Object {
            "contrastText": "#fff",
            "dark": "#303f9f",
            "light": "#7986cb",
            "main": "#3f51b5",
          },
          "secondary": Object {
            "contrastText": "#fff",
            "dark": "#c51162",
            "light": "#ff4081",
            "main": "#f50057",
          },
          "success": Object {
            "contrastText": "rgba(0, 0, 0, 0.87)",
            "dark": "#388e3c",
            "light": "#81c784",
            "main": "#4caf50",
          },
          "text": Object {
            "disabled": "rgba(0, 0, 0, 0.38)",
            "hint": "rgba(0, 0, 0, 0.38)",
            "primary": "rgba(0, 0, 0, 0.87)",
            "secondary": "rgba(0, 0, 0, 0.54)",
          },
          "tonalOffset": 0.2,
          "type": "light",
          "warning": Object {
            "contrastText": "rgba(0, 0, 0, 0.87)",
            "dark": "#f57c00",
            "light": "#ffb74d",
            "main": "#ff9800",
          },
        },
        "props": Object {},
        "shadows": Array [
          "none",
          "0px 2px 1px -1px rgba(0,0,0,0.2),0px 1px 1px 0px rgba(0,0,0,0.14),0px 1px 3px 0px rgba(0,0,0,0.12)",
          "0px 3px 1px -2px rgba(0,0,0,0.2),0px 2px 2px 0px rgba(0,0,0,0.14),0px 1px 5px 0px rgba(0,0,0,0.12)",
          "0px 3px 3px -2px rgba(0,0,0,0.2),0px 3px 4px 0px rgba(0,0,0,0.14),0px 1px 8px 0px rgba(0,0,0,0.12)",
          "0px 2px 4px -1px rgba(0,0,0,0.2),0px 4px 5px 0px rgba(0,0,0,0.14),0px 1px 10px 0px rgba(0,0,0,0.12)",
          "0px 3px 5px -1px rgba(0,0,0,0.2),0px 5px 8px 0px rgba(0,0,0,0.14),0px 1px 14px 0px rgba(0,0,0,0.12)",
          "0px 3px 5px -1px rgba(0,0,0,0.2),0px 6px 10px 0px rgba(0,0,0,0.14),0px 1px 18px 0px rgba(0,0,0,0.12)",
          "0px 4px 5px -2px rgba(0,0,0,0.2),0px 7px 10px 1px rgba(0,0,0,0.14),0px 2px 16px 1px rgba(0,0,0,0.12)",
          "0px 5px 5px -3px rgba(0,0,0,0.2),0px 8px 10px 1px rgba(0,0,0,0.14),0px 3px 14px 2px rgba(0,0,0,0.12)",
          "0px 5px 6px -3px rgba(0,0,0,0.2),0px 9px 12px 1px rgba(0,0,0,0.14),0px 3px 16px 2px rgba(0,0,0,0.12)",
          "0px 6px 6px -3px rgba(0,0,0,0.2),0px 10px 14px 1px rgba(0,0,0,0.14),0px 4px 18px 3px rgba(0,0,0,0.12)",
          "0px 6px 7px -4px rgba(0,0,0,0.2),0px 11px 15px 1px rgba(0,0,0,0.14),0px 4px 20px 3px rgba(0,0,0,0.12)",
          "0px 7px 8px -4px rgba(0,0,0,0.2),0px 12px 17px 2px rgba(0,0,0,0.14),0px 5px 22px 4px rgba(0,0,0,0.12)",
          "0px 7px 8px -4px rgba(0,0,0,0.2),0px 13px 19px 2px rgba(0,0,0,0.14),0px 5px 24px 4px rgba(0,0,0,0.12)",
          "0px 7px 9px -4px rgba(0,0,0,0.2),0px 14px 21px 2px rgba(0,0,0,0.14),0px 5px 26px 4px rgba(0,0,0,0.12)",
          "0px 8px 9px -5px rgba(0,0,0,0.2),0px 15px 22px 2px rgba(0,0,0,0.14),0px 6px 28px 5px rgba(0,0,0,0.12)",
          "0px 8px 10px -5px rgba(0,0,0,0.2),0px 16px 24px 2px rgba(0,0,0,0.14),0px 6px 30px 5px rgba(0,0,0,0.12)",
          "0px 8px 11px -5px rgba(0,0,0,0.2),0px 17px 26px 2px rgba(0,0,0,0.14),0px 6px 32px 5px rgba(0,0,0,0.12)",
          "0px 9px 11px -5px rgba(0,0,0,0.2),0px 18px 28px 2px rgba(0,0,0,0.14),0px 7px 34px 6px rgba(0,0,0,0.12)",
          "0px 9px 12px -6px rgba(0,0,0,0.2),0px 19px 29px 2px rgba(0,0,0,0.14),0px 7px 36px 6px rgba(0,0,0,0.12)",
          "0px 10px 13px -6px rgba(0,0,0,0.2),0px 20px 31px 3px rgba(0,0,0,0.14),0px 8px 38px 7px rgba(0,0,0,0.12)",
          "0px 10px 13px -6px rgba(0,0,0,0.2),0px 21px 33px 3px rgba(0,0,0,0.14),0px 8px 40px 7px rgba(0,0,0,0.12)",
          "0px 10px 14px -6px rgba(0,0,0,0.2),0px 22px 35px 3px rgba(0,0,0,0.14),0px 8px 42px 7px rgba(0,0,0,0.12)",
          "0px 11px 14px -7px rgba(0,0,0,0.2),0px 23px 36px 3px rgba(0,0,0,0.14),0px 9px 44px 8px rgba(0,0,0,0.12)",
          "0px 11px 15px -7px rgba(0,0,0,0.2),0px 24px 38px 3px rgba(0,0,0,0.14),0px 9px 46px 8px rgba(0,0,0,0.12)",
        ],
        "shape": Object {
          "borderRadius": 4,
        },
        "spacing": [Function],
        "transitions": Object {
          "create": [Function],
          "duration": Object {
            "complex": 375,
            "enteringScreen": 225,
            "leavingScreen": 195,
            "short": 250,
            "shorter": 200,
            "shortest": 150,
            "standard": 300,
          },
          "easing": Object {
            "easeIn": "cubic-bezier(0.4, 0, 1, 1)",
            "easeInOut": "cubic-bezier(0.4, 0, 0.2, 1)",
            "easeOut": "cubic-bezier(0.0, 0, 0.2, 1)",
            "sharp": "cubic-bezier(0.4, 0, 0.6, 1)",
          },
          "getAutoHeightDuration": [Function],
        },
        "typography": Object {
          "body1": Object {
            "fontFamily": "HelveticaNeue,Roboto,Arial",
            "fontSize": "1rem",
            "fontWeight": 400,
            "lineHeight": 1.5,
          },
          "body2": Object {
            "fontFamily": "HelveticaNeue,Roboto,Arial",
            "fontSize": "0.875rem",
            "fontWeight": 400,
            "lineHeight": 1.43,
          },
          "button": Object {
            "fontFamily": "HelveticaNeue,Roboto,Arial",
            "fontSize": "0.875rem",
            "fontWeight": 500,
            "lineHeight": 1.75,
            "textTransform": "uppercase",
          },
          "caption": Object {
            "fontFamily": "HelveticaNeue,Roboto,Arial",
            "fontSize": "0.75rem",
            "fontWeight": 400,
            "lineHeight": 1.66,
          },
          "fontFamily": "HelveticaNeue,Roboto,Arial",
          "fontSize": 14,
          "fontWeightBold": 700,
          "fontWeightLight": 300,
          "fontWeightMedium": 500,
          "fontWeightRegular": 400,
          "h1": Object {
            "fontFamily": "HelveticaNeue,Roboto,Arial",
            "fontSize": "6rem",
            "fontWeight": 300,
            "lineHeight": 1.167,
          },
          "h2": Object {
            "fontFamily": "HelveticaNeue,Roboto,Arial",
            "fontSize": "3.75rem",
            "fontWeight": 300,
            "lineHeight": 1.2,
          },
          "h3": Object {
            "fontFamily": "HelveticaNeue,Roboto,Arial",
            "fontSize": "3rem",
            "fontWeight": 400,
            "lineHeight": 1.167,
          },
          "h4": Object {
            "fontFamily": "HelveticaNeue,Roboto,Arial",
            "fontSize": "2.125rem",
            "fontWeight": 400,
            "lineHeight": 1.235,
          },
          "h5": Object {
            "fontFamily": "HelveticaNeue,Roboto,Arial",
            "fontSize": "1.5rem",
            "fontWeight": 400,
            "lineHeight": 1.334,
          },
          "h6": Object {
            "fontFamily": "HelveticaNeue,Roboto,Arial",
            "fontSize": "1.25rem",
            "fontWeight": 500,
            "lineHeight": 1.6,
          },
          "htmlFontSize": 16,
          "overline": Object {
            "fontFamily": "HelveticaNeue,Roboto,Arial",
            "fontSize": "0.75rem",
            "fontWeight": 400,
            "lineHeight": 2.66,
            "textTransform": "uppercase",
          },
          "pxToRem": [Function],
          "round": [Function],
          "subtitle1": Object {
            "fontFamily": "HelveticaNeue,Roboto,Arial",
            "fontSize": "1rem",
            "fontWeight": 400,
            "lineHeight": 1.75,
          },
          "subtitle2": Object {
            "fontFamily": "HelveticaNeue,Roboto,Arial",
            "fontSize": "0.875rem",
            "fontWeight": 500,
            "lineHeight": 1.57,
          },
        },
        "zIndex": Object {
          "appBar": 1100,
          "drawer": 1200,
          "mobileStepper": 1000,
          "modal": 1300,
          "snackbar": 1400,
          "speedDial": 1050,
          "tooltip": 1500,
        },
      }
    }
  >
    <ApolloProvider
      client={
        ApolloClient {
          "cache": InMemoryCache {
            "addTypename": true,
            "cacheKeyRoot": KeyTrie {
              "weakness": true,
            },
            "config": Object {
              "addTypename": true,
              "dataIdFromObject": [Function],
              "fragmentMatcher": HeuristicFragmentMatcher {},
              "freezeResults": false,
              "resultCaching": true,
            },
            "data": DepTrackingCache {
              "data": Object {},
              "depend": [Function],
            },
            "maybeBroadcastWatch": [Function],
            "optimisticData": DepTrackingCache {
              "data": Object {},
              "depend": [Function],
            },
            "silenceBroadcast": false,
            "storeReader": StoreReader {
              "executeSelectionSet": [Function],
              "executeStoreQuery": [Function],
              "executeSubSelectedArray": [Function],
              "freezeResults": false,
            },
            "storeWriter": StoreWriter {},
            "typenameDocumentCache": Map {},
            "watches": Set {},
          },
          "clearStoreCallbacks": Array [],
          "defaultOptions": Object {
            "watchQuery": Object {
              "errorPolicy": "all",
              "fetchPolicy": "cache-first",
            },
          },
          "disableNetworkFetches": false,
          "link": ApolloLink {
            "request": [Function],
          },
          "localState": LocalState {
            "cache": InMemoryCache {
              "addTypename": true,
              "cacheKeyRoot": KeyTrie {
                "weakness": true,
              },
              "config": Object {
                "addTypename": true,
                "dataIdFromObject": [Function],
                "fragmentMatcher": HeuristicFragmentMatcher {},
                "freezeResults": false,
                "resultCaching": true,
              },
              "data": DepTrackingCache {
                "data": Object {},
                "depend": [Function],
              },
              "maybeBroadcastWatch": [Function],
              "optimisticData": DepTrackingCache {
                "data": Object {},
                "depend": [Function],
              },
              "silenceBroadcast": false,
              "storeReader": StoreReader {
                "executeSelectionSet": [Function],
                "executeStoreQuery": [Function],
                "executeSubSelectedArray": [Function],
                "freezeResults": false,
              },
              "storeWriter": StoreWriter {},
              "typenameDocumentCache": Map {},
              "watches": Set {},
            },
            "client": [Circular],
          },
          "mutate": [Function],
          "query": [Function],
          "queryDeduplication": true,
          "queryManager": QueryManager {
            "assumeImmutableResults": false,
            "clientAwareness": Object {
              "name": undefined,
              "version": undefined,
            },
            "dataStore": DataStore {
              "cache": InMemoryCache {
                "addTypename": true,
                "cacheKeyRoot": KeyTrie {
                  "weakness": true,
                },
                "config": Object {
                  "addTypename": true,
                  "dataIdFromObject": [Function],
                  "fragmentMatcher": HeuristicFragmentMatcher {},
                  "freezeResults": false,
                  "resultCaching": true,
                },
                "data": DepTrackingCache {
                  "data": Object {},
                  "depend": [Function],
                },
                "maybeBroadcastWatch": [Function],
                "optimisticData": DepTrackingCache {
                  "data": Object {},
                  "depend": [Function],
                },
                "silenceBroadcast": false,
                "storeReader": StoreReader {
                  "executeSelectionSet": [Function],
                  "executeStoreQuery": [Function],
                  "executeSubSelectedArray": [Function],
                  "freezeResults": false,
                },
                "storeWriter": StoreWriter {},
                "typenameDocumentCache": Map {},
                "watches": Set {},
              },
            },
            "fetchQueryRejectFns": Map {},
            "idCounter": 1,
            "inFlightLinkObservables": Map {},
            "link": ApolloLink {
              "request": [Function],
            },
            "localState": LocalState {
              "cache": InMemoryCache {
                "addTypename": true,
                "cacheKeyRoot": KeyTrie {
                  "weakness": true,
                },
                "config": Object {
                  "addTypename": true,
                  "dataIdFromObject": [Function],
                  "fragmentMatcher": HeuristicFragmentMatcher {},
                  "freezeResults": false,
                  "resultCaching": true,
                },
                "data": DepTrackingCache {
                  "data": Object {},
                  "depend": [Function],
                },
                "maybeBroadcastWatch": [Function],
                "optimisticData": DepTrackingCache {
                  "data": Object {},
                  "depend": [Function],
                },
                "silenceBroadcast": false,
                "storeReader": StoreReader {
                  "executeSelectionSet": [Function],
                  "executeStoreQuery": [Function],
                  "executeSubSelectedArray": [Function],
                  "freezeResults": false,
                },
                "storeWriter": StoreWriter {},
                "typenameDocumentCache": Map {},
                "watches": Set {},
              },
              "client": [Circular],
            },
            "mutationStore": MutationStore {
              "store": Object {},
            },
            "onBroadcast": [Function],
            "pollingInfoByQueryId": Map {},
            "queries": Map {},
            "queryDeduplication": true,
            "queryStore": QueryStore {
              "store": Object {},
            },
            "ssrMode": false,
            "transformCache": WeakMap {},
          },
          "reFetchObservableQueries": [Function],
          "resetStore": [Function],
          "resetStoreCallbacks": Array [],
          "store": DataStore {
            "cache": InMemoryCache {
              "addTypename": true,
              "cacheKeyRoot": KeyTrie {
                "weakness": true,
              },
              "config": Object {
                "addTypename": true,
                "dataIdFromObject": [Function],
                "fragmentMatcher": HeuristicFragmentMatcher {},
                "freezeResults": false,
                "resultCaching": true,
              },
              "data": DepTrackingCache {
                "data": Object {},
                "depend": [Function],
              },
              "maybeBroadcastWatch": [Function],
              "optimisticData": DepTrackingCache {
                "data": Object {},
                "depend": [Function],
              },
              "silenceBroadcast": false,
              "storeReader": StoreReader {
                "executeSelectionSet": [Function],
                "executeStoreQuery": [Function],
                "executeSubSelectedArray": [Function],
                "freezeResults": false,
              },
              "storeWriter": StoreWriter {},
              "typenameDocumentCache": Map {},
              "watches": Set {},
            },
          },
          "typeDefs": undefined,
          "version": "2.6.10",
          "watchQuery": [Function],
        }
      }
    >
      <div
        className="content"
      >
        <Route
          path="/"
          render={[Function]}
        />
      </div>
      <Footer />
    </ApolloProvider>
  </ThemeProvider>
</StylesProvider>
`;
